analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -59,6 +60,71 @@ public CSharpSymbolUsageCollector(Compilation compilation, IEnumerable<ISymbol>
             knownSymbolNames = knownSymbols.Select(GetName).ToHashSet();
         }
 
+        // TupleExpression "(a, b) = qix"
+        // ParenthesizedVariableDesignation "var (a, b) = quix" inside a DeclarationExpression
+        public override void VisitAssignmentExpression(AssignmentExpressionSyntax node)
+        {
+            var assignmentLeft = node.Left;
+            Func<int> getLeftTupleCount = null;
+            // in a newer Roslyn API, we could check the Operation to be DeconstructionAssignment
```

Comment:
pragmatically, I will stop here with this implementation (to avoid going into refactoring rabbit hole)

Indeed, we can use this. I'm not sure how expensive [GetOperation](https://github.com/dotnet/roslyn/blob/4929b2e60abbdbb6cbe72c856d6f370c8cf13b35/src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs#L483) is,